cardinal/gamestate/Storage.go

```diff
@@ -0,0 +1,24 @@
+package gamestate
+
+import (
+	"context"
+)
+
+// Storage is the interface for all available stores related to the game loop
+// there is another store like interface for other logistical values located in `ecs.metastorage`
+type Storage interface {
```

Comment:
Renaming an existing struct/interface, so that the newly freed up name can be used in a new concept is really confusing when trying to review code. Call this "PrimitiveStorage" and revert the "MetaStroage" back to what it was originally.

After the new functionality is in, if we're unhappy with the names we can make a PR that only changes the names.

Reply:
Hmmm I think we both agree that it Both "storages" need different names. But in my opinion the gamestate is the "primary" storage here and the other "storage" is more secondary. So I would just rename the other storage instead of the one in gamestate. I can do the opposite of what you suggest here, changing metastorage to primitivestorage. Would that work? 

Honestly though I don't care too much about the naming here, I can still use your suggestion here, but I present the one above as an alternative I'm a bit more in favor of, and if you agree we can go with that. If not I'll go with your initial suggestion, not a huge deal to me. Lmk.

Reply:
In this PR I'd prefer to leave any already-existing interfaces, strcuts, packages as-is, and make new names for whatever new things you're introducing. I think that's limited to the new interface in "Stroage.go" as well as the redis implementation in "cardinal/gamestate/redis.go". I suggest naming those PrimitiveStorage and RedisPrimitativeStorage and reverting the other names.